notifications/file_event_notifications.py

```diff
@@ -0,0 +1,321 @@
+"""Event system + File events
```

Comment:
This take the old and mostly underutilized `Event` class and reduces it's complecity so it only handles files notifications.